noxfile.py

```diff
@@ -37,17 +49,23 @@ def integration_tests(session: nox.Session) -> None:
     """Run the integration tests."""
     session.install("--editable", ".[test]")
     session.run(
-        "scripts/integration-test.sh",
+        "pytest",
+        "tests/integration",
+        "-rxXs",  # Show provided reason in summary for (x)fail, (X)pass, and (s)kipped tests
         *session.posargs,
         env=dict(
             EARTHDATA_USERNAME=os.environ["EARTHDATA_USERNAME"],
             EARTHDATA_PASSWORD=os.environ["EARTHDATA_PASSWORD"],
         ),
         external=True,
+        # NOTE: integration test are permitted to pass if the failure rate was less than a hardcoded threshold.
+        #       PyTest will return 99 if there were some failures, but less than the threshold. For more details, see:
+        #       `pytest_sessionfinish` in tests/integration/conftest.py
+        success_codes=[0, 99],
```

Comment:
Don't we also need to replace the hard-coded `99` with the value of `EARTHACCESS_ALLOWABLE_FAILURE_STATUS_CODE` here too (defaulting to 99, if not set)?

Reply:
I don't think so here because we're not passing that environment down to the test environment, so it should always use the default `99`.:
https://github.com/nsidc/earthaccess/pull/872/files#diff-f7a16a65f061822bcc73b8296f4dc837353d379d8d9cc5307982cb6941442835R56-R59

Reply:
I guess that's why I'm a bit confused. Where/when/how do you expect to set the var and have it used? You mentioned something about running tests from your IDE, so I guess I'm wondering how that would work (I just run them from the terminal, and haven't attempted running from my IDE).

Reply:
Yes, it's only useful for directly invoking `pytest tests/integration ...` either in the terminal or via your IDE so that you get a successful status code. If you invoke via nox, it handles that bit for you, but making VS Code and PyCharm aware of nox is significantly harder than making them aware of pytest. 

Preview of the upcoming PR where I go into it a bit in the context of an IDE:
https://github.com/nsidc/earthaccess/blob/dev-guide/docs/contributing/development.md#ide-setup

Reply:
I'm going to merge this one and open the next one so we can continue the conversation there.

(ideally, we wouldn't have flakey tests...)